net/net-memcache-server.cpp

```diff
@@ -771,64 +773,69 @@ int return_one_key_list (struct connection *c, const char *key, int key_len, int
     if (ptr >= s) {
       advance_write_ptr (&c->Out, ptr - (s - 480));
       ptr = get_write_ptr (&c->Out, 512);
-      if (!ptr) return -1;
+      if (!ptr) {
+        return -1;
+      }
       s = ptr + 480;
     }
     if (mode >= 0) {
       if (i || res != 0x7fffffff) {
         *ptr++ = ',';  w++;
       }
-      w += t = sprintf (ptr, "%d", R[i]);
+      w += t = snprintf(ptr, 10, "%d", R[i]);
```

Comment:
10 is definitely not enough
"%d" is 10 digits + potential minus symbol + null terminated byte, in total 12 bytes at least.